solutions/hw2/task2/cacher.py

```diff
@@ -0,0 +1,62 @@
+from typing import List, Any, Tuple, Callable
+import functools
+
+
+class FunctionCallCache:
+    nameless_args: Tuple
+    named_args: dict
+    result: Any
+
+    def __init__(self, args: Tuple, kwargs: dict, result):
+        self.nameless_args = args
+        self.named_args = kwargs
```

Comment:
I am not sure about this. Are you sure that it works correctly? because dict sores unordered items

Reply:
What's wrong with storing named arguments for one function call in unordered dict?